src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -0,0 +1,3187 @@
+// are concerned.
+
+namespace System.Text.RegularExpressions.Generator
+{
+    public partial class RegexGenerator
+    {
+        /// <summary>Code for a [GeneratedCode] attribute to put on the top-level generated members.</summary>
+        private static readonly string s_generatedCodeAttribute = $"[global::System.CodeDom.Compiler.GeneratedCodeAttribute(\"{typeof(RegexGenerator).Assembly.GetName().Name}\", \"{typeof(RegexGenerator).Assembly.GetName().Version}\")]";
+        /// <summary>Header comments and usings to include at the top of every generated file.</summary>
+        private static readonly string[] s_headers = new string[]
+        {
+            "// <auto-generated/>",
+            "#nullable enable",
+            "#pragma warning disable CS0162 // Unreachable code",
+            "#pragma warning disable CS0164 // Unreferenced label",
+            "#pragma warning disable CS0168 // Variable declared but never used",
+            "#pragma warning disable CS0219 // Variable assigned but never used",
+            "",
+        };
+
+        /// <summary>Generates the code for one regular expression class.</summary>
+        private static string EmitRegexType(RegexType regexClass)
+        {
+            var sb = new StringBuilder(1024);
+            var writer = new IndentedTextWriter(new StringWriter(sb));
+
+            // Emit the namespace
+            if (!string.IsNullOrWhiteSpace(regexClass.Namespace))
+            {
+                writer.WriteLine($"namespace {regexClass.Namespace}");
+                writer.WriteLine("{");
+                writer.Indent++;
+            }
+
+            // Emit containing types
+            RegexType parent = regexClass.ParentClass;
+            var parentClasses = new Stack<string>();
+            while (parent != null)
+            {
+                parentClasses.Push($"partial {parent.Keyword} {parent.Name} {parent.Constraints}");
+                parent = parent.ParentClass;
+            }
+            while (parentClasses.Count != 0)
+            {
+                writer.WriteLine($"{parentClasses.Pop()}");
+                writer.WriteLine("{");
+                writer.Indent++;
+            }
+
+            // Emit the direct parent type
+            writer.WriteLine($"partial {regexClass.Keyword} {regexClass.Name} {regexClass.Constraints}");
+            writer.WriteLine("{");
+            writer.Indent++;
+
+            // Generate a unique name to describe the regex instance.  This includes both
+            // the method name the user provided and a unique counter value, plus a simple
+            // non-randomized (for determinism) hash of the previous content to try to make
+            // the name that much harder to predict.
+            string generatedName = $"GeneratedRegex_{regexClass.Method.MethodName}_";
+            generatedName += ComputeStringHash(generatedName).ToString("X");
```

Comment:
I don't get where the unique counter value comes from?

Reply:
The comment is stale.  I fixed it.